python/tests/test_stdlib.py

```diff
@@ -19,8 +19,8 @@ def test_text_ops(self):
         #self.assertEqualMettaRunnerResults(metta.run('!(parse "$X")'),
         #                                  [[(V("X"))]])
 
-        self.assertEqualMettaRunnerResults(metta.run('!(parse "\\"A\\"")'),
-                                           [[(ValueAtom("A"))]])
+        self.assertEqualMettaRunnerResults(repr(metta.run('!(parse "\\"A\\"")')),
+                                           '[[\\"A\\"]]')
```

Comment:
1. Test should check which atoms are returned by `parse` not what their string representation is
2. At the moment test is passing in the following form:
```
        self.assertEqualMettaRunnerResults(metta.run('!(parse "\\"A\\"")'),
                                           [[(S('\\"A\\"'))]])
```
One can see that now `parse` returns symbol instead of string. It is because `parse` parses `str(atom)` instead of internal string. In this particular case atom is a string `"A"` but `str()` from this atom gives `\"A\"` (this is what is changed by the PR). This sequence is parsed as symbol `\"A\"` because escape sequences are processed by MeTTa parser only inside strings and `\"A\"` is not a string because it starts from `\` symbol. Thus the result of the test is unexpected. To make it proper we need to change `parse` implementation to work with original string not with the result of `str()`. Test should be passed in its original form:
```suggestion
        self.assertEqualMettaRunnerResults(metta.run('!(parse "\\"A"\\")'),
                                           [[(ValueAtom("A"))]])
```

Reply:
To properly solve it, I would suggest change `parse` implementation to extract Python string from grounded atom on input. Should be as simple as calling `.get_object()` on an input atom and checking that this is a really string.